File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        // we will directly use two pointer here as elements are soted
        int i=0,j=numbers.size()-1;
        while(i<j){
            int sum = numbers[i]+numbers[j];
            if(sum==target){
                return {i+1,j+1};
            }
            if(sum>target){
                j--;
            }
            else i++;
        }
        return {-1,-1};
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        mp = {}
        for i in range(0,len(numbers)):
            if target-numbers[i] in mp:
                return [mp[target-numbers[i]],i+1]
            mp[numbers[i]]=i+1
        return []
        
           

            
        